modules/core/src/main/java/org/apache/fluo/core/util/ScanUtil.java

```diff
@@ -128,24 +125,20 @@ public static void scanFluo(ScanOpts options, FluoConfiguration sConfig) {
             sb.append(rcv.getsValue());
             System.out.println(sb.toString());
           }
-          entriesFound++;
+
           if (System.out.checkError()) {
             break;
           }
         }
 
-        if (entriesFound == 0) {
-          System.out.println("\nNo data found\n");
-        }
       } catch (FluoException e) {
-        System.out.println("Scan failed - " + e.getMessage());
+        System.err.println("Scan failed - " + e.getMessage());
+        System.exit(-1);
```

Comment:
Wouldn't it be better to just throw the exception so the user has the full stack trace? Throwing will automatically result in an appropriate non-zero exit code. Plus, avoiding the call to `System.exit` makes this app a bit more friendly to any future (or external) work which might try to call this code, such as in a GUI or other multi-threaded app.

Also, you should avoid negative exit codes. They don't work well cross-platform, so it's generally bad practice. It's probably not going to cause problems on any platforms where Fluo is run, but it's a good habit to break.

I think the best thing is to just throw the exception.

Reply:
That sounds fine.  I changed this code to make it consistent with other existing code. I think it would be best to make this change in another PR where all of the code is made consistent.

```bash
$ find . -name "*.java" | xargs grep System.exit
./src/main/java/org/apache/fluo/command/BaseOpts.java:      System.exit(-1);
./src/main/java/org/apache/fluo/command/BaseOpts.java:      System.exit(1);
./src/main/java/org/apache/fluo/command/CommandUtil.java:      System.exit(-1);
./src/main/java/org/apache/fluo/command/CommandUtil.java:        System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoInit.java:      System.exit(0);
./src/main/java/org/apache/fluo/command/FluoInit.java:      System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoInit.java:      System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoInit.java:        System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoInit.java:        System.exit(0);
./src/main/java/org/apache/fluo/command/FluoInit.java:            System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoInit.java:            System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoInit.java:        System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoExec.java:      System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoWait.java:          System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoWait.java:      System.exit(-1);
./src/main/java/org/apache/fluo/command/FluoWait.java:      System.exit(-1);
```

Reply:
Opened #983

Reply:
If it's going to be fixed in another issue, then I'm okay with it here (as long as this is merged in first), but I don't see much point in adding something new here that we know will get removed.

Reply:
> that part of the changeset isn't necessary to accomplish the stated goal of this issue.

It is unrelated.  Its a bug I noticed while working this issue.  The current code will not set an error return code for the process.